packages/utils/test/ratelimit.test.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  DEFAULT_RETRY_AFTER,
+  disabledUntil,
+  isRateLimited,
+  parseRetryAfterHeader,
+  RateLimits,
+  updateRateLimits,
+} from '../src/ratelimit';
+
+describe('parseRetryAfterHeader()', () => {
+  test('should fallback to 60s when incorrect header provided', () => {
+    expect(parseRetryAfterHeader('x')).toEqual(DEFAULT_RETRY_AFTER);
+  });
+
+  test('should correctly parse delay-based header', () => {
+    expect(parseRetryAfterHeader('1337')).toEqual(1337 * 1000);
+  });
+
+  test('should correctly parse date-based header', () => {
+    expect(
+      parseRetryAfterHeader('Wed, 21 Oct 2015 07:28:13 GMT', new Date('Wed, 21 Oct 2015 07:28:00 GMT').getTime()),
+    ).toEqual(13 * 1000);
+  });
+});
+
+describe('disabledUntil()', () => {
+  test('should return 0 when no match', () => {
+    expect(disabledUntil({}, 'error')).toEqual(0);
+  });
+
+  test('should return matched value', () => {
+    expect(disabledUntil({ error: 42 }, 'error')).toEqual(42);
+  });
+
+  test('should fallback to `all` category', () => {
+    expect(disabledUntil({ all: 42 }, 'error')).toEqual(42);
+  });
+});
+
+describe('isRateLimited()', () => {
+  test('should return false when no match', () => {
+    expect(isRateLimited({}, 'error')).toEqual(false);
+  });
+
+  test('should return false when matched value is in the past', () => {
+    expect(isRateLimited({ error: 10 }, 'error', 42)).toEqual(false);
+  });
+
+  test('should return true when matched value is in the future', () => {
+    expect(isRateLimited({ error: 50 }, 'error', 42)).toEqual(true);
+  });
+
+  test('should fallback to the `all` category when given one is not matched', () => {
+    expect(isRateLimited({ all: 10 }, 'error', 42)).toEqual(false);
+    expect(isRateLimited({ all: 50 }, 'error', 42)).toEqual(true);
+  });
+});
+
+describe('updateRateLimits()', () => {
+  test('should return same limits when no headers provided', () => {
+    const rateLimits: RateLimits = {
+      error: 42,
+      transaction: 1337,
+    };
+    const headers = {};
+    const updatedRateLimits = updateRateLimits(rateLimits, headers);
+    expect(rateLimits).toStrictEqual(updatedRateLimits);
```

Comment:
```suggestion
    expect(updatedRateLimits).toStrictEqual(rateLimits);
```

Our expectation is, effectively, "the result will be correct," so I think it makes more sense to swap these. Also, I'm curious why you're using strict equal here over equal. Are you testing that they didn't get converted to strings?

Reply:
`updatedRateLimits` and `rateLimits` should be rate limits objects. I guess we don't need the `undefined `checks from `toStrictEqual`, I'll just update it